### DISCUSSION TITLE
Use only deploy hooks, instead of deploy on push.

### DISCUSSION THREAD

**Discussion by yassinebridi** [vercel/community#286] [Show & Tell]
I want to trigger a vercel deployment from github actions, so i'm using a deploy hook to do that. But even if i don't trigger a deployment using the hook, vercel deploys the project anyway.

Vercel have no option to use only the deploy hook.

**Comment by cansin** [comment]
I believe you can rely on https://vercel.com/docs/configuration#git/github-enabled

```
{
  "github": {
    "enabled": false
  }
}
```

at `vercel.json`

**Reply by yassinebridi** [reply]
This unfortunately disables Deploy Hooks too, which i'm using to deploy my app manually.

**Comment by JohnDeved** [comment]
have you found a solution for this? @yassinebridi

**Reply by yassinebridi** [reply]
Still nothing, i'm using a GitHub action to deploy at the moment, but Vercel still triggers another deploy when pushing.
Hey @styfle, i wonder if this is possible at the moment?

**Reply by styfle** [reply]
Deploy hooks are designed for CMS changes if your site is completely static so `git push` will always deploy code changes and the deploy hook can deploy content changes. 

Read more: https://vercel.com/docs/more/deploy-hooks

**Reply by styfle** [reply]
Is there a reason why you're using GitHub Actions to deploy instead of relying on `git push`?

**Reply by yassinebridi** [reply]
There is two reasons actually:
- I'm running a script that determines what packages changed, and then decide which GitHub action steps to be fired.
- For each successful job, i trigger a notification to a 3rd party service.

**Reply by styfle** [reply]
You can utilize that script in the [Ignore Build Step](https://vercel.com/support/articles/how-do-i-use-the-ignored-build-step-field-on-vercel) so that Vercel knows if it should ignore the `git push` or deploy.

**Reply by sarahdayan** [reply]
@styfle Is there a way to pass arbitrary data when using Deploy Hooks so that there's a way to know where a deploy trigger is coming from, and bypass the restrictions from the Ignore Build Step script? I also want to disable deploy on push, but so far I haven't found a way to script my ignore step so it cancels builds on push and allows them when hitting the hook.

**Comment by evaldofelipe** [comment]
+1

**Comment by remjx** [comment]
as @styfle mentioned you should be able to accomplish it with an [Ignore Build Step](https://vercel.com/support/articles/how-do-i-use-the-ignored-build-step-field-on-vercel) script

**Comment by thejuan** [comment]
We are also looking for this. We don't want to migrate our CI to vercel yet and want to manually trigger a deploy once the tests run green.

**Reply by JohnDeved** [reply]
its possible to make use of the vercel cli to archive custom CI deployments.
check this source code https://github.com/amondnet/vercel-action

**Comment by Pieparker** [comment]
For posterity, [here's a great little tidbit](https://github.com/vercel/community/discussions/285#discussioncomment-1696833) from a similar thread.

Due to the overlap/to prevent duplicate discussions, I will lock this thread in favour of https://github.com/vercel/community/discussions/285.
